localstack/services/sts/provider.py

```diff
@@ -14,3 +31,29 @@ def get_caller_identity(self, context: RequestContext, **kwargs) -> GetCallerIde
         if "user/moto" in response["Arn"] and "sts" in response["Arn"]:
             response["Arn"] = f"arn:aws:iam::{response['Account']}:root"
         return response
+
+    def assume_role(
+        self,
+        context: RequestContext,
+        role_arn: arnType,
+        role_session_name: roleSessionNameType,
+        policy_arns: policyDescriptorListType = None,
+        policy: unrestrictedSessionPolicyDocumentType = None,
+        duration_seconds: roleDurationSecondsType = None,
+        tags: tagListType = None,
+        transitive_tag_keys: tagKeyListType = None,
+        external_id: externalIdType = None,
+        serial_number: serialNumberType = None,
+        token_code: tokenCodeType = None,
+        source_identity: sourceIdentityType = None,
+        provided_contexts: ProvidedContextsListType = None,
+        **kwargs,
+    ) -> AssumeRoleResponse:
+        response: AssumeRoleResponse = call_moto(context)
+
+        if tags:
+            transformed_tags = {tag["Key"]: tag["Value"] for tag in tags}
+            store = sts_stores[context.account_id][context.region]
```

Comment:
If I understand this correctly, it means this feature is not cross-account compatible for now, since we should save in the target account/region (based on the ARN of the role).

Assume_role is probably one of the most used operations in the context of cross-account usage :thinking: